src/main/mpack/common-services/SOLR/7.3.1/metainfo.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+<!--
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for thparae specific language governing permissions and
+   limitations under the License.
+-->
+<metainfo>
+    <schemaVersion>2.0</schemaVersion>
+    <services>
+        <service>
+            <name>SOLR</name>
+            <version>7.3.1</version>
```

Comment:
[Q] I chose to base the 7.3.1 service version off of 6.6.2.  So I _think_ I should be able to take advantage of that inheritance and really trim down this PR, since many of the files appear identically over in the 6.6.2 directory structure.

But confusingly, I noticed that 5.5.5 and 6.6.2 have a similar relationship.  That is `common-services/6.6.2` extends `common-services/5.5.5`, but the two file trees share a good number of files that match identically.  Does the “service inheritance” not share as much as I think it does, or am I misunderstanding it?  Or are we intentionally trying to not lean on the inheritance too much for some reason?

(The following `common-services` files all appear to be identical in 5.5.5 and 6.6.2:
```
➜  SOLR git:(branch_7x) cd src/main/mpack/common-services/SOLR && diff -srq 5.5.5/ 6.6.2/ | grep identical | awk '{print $2}'
5.5.5/package/scripts/service_check.py
5.5.5/package/scripts/setup_solr_cloud.py
5.5.5/package/scripts/setup_solr_hdfs_support.py
5.5.5/package/scripts/setup_solr_ssl_support.py
5.5.5/package/scripts/solr.py
5.5.5/package/scripts/solr_utils.py
5.5.5/package/scripts/status_params.py
5.5.5/package/templates/log4j2.xml
5.5.5/package/templates/solr_metrics_jaas.conf.j2
5.5.5/package/templates/solr_server_jaas.conf.j2
```
)

Reply:
Yes, the `packages` behaves a bit different from the rest. 

> The following directories if defined in the current service version replace those from the parent service version:
> packages
> upgrades
> This means the files included in those directories at the parent level will not be inherited.  You will need to copy all the files you wish to keep from that directory structure. 

The rest of folder are "merge" (there are some setting that we can use, in order to define how they are merge.)

more info here: https://cwiki.apache.org/confluence/display/AMBARI/Service+Inheritance